app/routes/members-area.my-account.update-details/route.tsx

```diff
@@ -0,0 +1,26 @@
+import { ActionFunctionArgs, redirect } from '@remix-run/node';
+import { initializeEcomApiForRequest } from '~/src/wix/ecom/session';
+
+export async function action({ request }: ActionFunctionArgs) {
+    const api = await initializeEcomApiForRequest(request);
+    const formData = await request.formData();
+
+    const userId = formData.get('userId') as string | undefined;
+    const firstName = formData.get('firstName') as string | undefined;
+    const lastName = formData.get('lastName') as string | undefined;
+    const phoneNumber = formData.get('phoneNumber') as string | undefined;
+
+    if (!userId) {
+        throw new Error('Missing user id');
+    }
+
+    await api.updateUser(userId, {
+        contact: {
+            firstName,
+            lastName,
+            phones: phoneNumber ? [phoneNumber] : [],
+        },
+    });
+
+    return redirect('/members-area/my-account');
```

Comment:
do we need this redirect? is it really reloads page? I think it is better to just do nothing

Reply:
it might be an issue. when I was working on that, I found that sometimes phone number is not updated immediately. could you please double check this?

Reply:
otherwise we'll have `/members-area/my-account/update-details` in the url bar

Reply:
and also, why do we need to fetch new data? I mean, new data will be the same as we posted, so the same that is current input values, doesn't it?

Reply:
so, it works, because after redirect loader data is updated, but it is used on our page as initial value for `setState`, so chaning that has no effect on a `useState`